cunumeric/array.py

```diff
@@ -1741,6 +1741,63 @@ def astype(
         result._thunk.convert(self._thunk, warn=False)
         return result
 
+    def take(self, indices, axis=None, out=None, mode="raise"):
+        if not np.isscalar(indices):
+            # if indices is a tuple or list, bring sub-tuples to the same shape
+            # and concatenate them
+            indices = convert_to_cunumeric_ndarray(indices)
+
+        if axis is None:
+            self = self.ravel()
+            axis = 0
+        elif axis < 0:
+            axis = self.ndim + axis
+        elif axis >= self.ndim:
+            raise ValueError("axis is out of bounds for array of dimension")
+
+        if mode == "raise":
```

Comment:
NumPy allows negative indices in this case, e.g. for `self.shape[axis] == 2` it accepts `-1` and `-2`.